databricks/koalas/groupby.py

```diff
@@ -0,0 +1,159 @@
+#
+# Copyright (C) 2019 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+A wrapper for GroupedData to behave similar to pandas GroupBy.
+"""
+
+from functools import partial
+from typing import Any, List, Union
+
+from pyspark.sql import functions as F
+from pyspark.sql.types import FloatType, DoubleType, NumericType
+
+from databricks.koalas.frame import DataFrame
+from databricks.koalas.metadata import Metadata
+from databricks.koalas.missing.groupby import _MissingPandasLikeDataFrameGroupBy, \
+    _MissingPandasLikeSeriesGroupBy
+from databricks.koalas.series import Series, _col
+
+
+ColumnLike = Union[str, Series]
+
+
+class GroupBy(object):
+
+    def __new__(cls, obj: Union[DataFrame, Series], *args, **kwargs):
+        if isinstance(obj, DataFrame):
+            return super(GroupBy, cls).__new__(DataFrameGroupBy)
+        elif isinstance(obj, Series):
+            return super(GroupBy, cls).__new__(SeriesGroupBy)
+        else:
+            raise TypeError('invalid type: {}'.format(obj))
```

Comment:
in addition to pointing out it is invalid, say we only allow DataFrame or Series.